client/fcidrClient.sp

```diff
@@ -0,0 +1,110 @@
+#include "SCR-Version"
+
+#include <sourcemod>
+#include <socket>
+#include <bytebuff>
+
+#define PROTOCOL_VERSION 1
+
+
+/**
+ * Base message structure
+ * 
+ */
+methodmap Header < ByteBuffer
+{
+	public Header()
+	{
+		return view_as<Header>(CreateByteBuffer());
+	}
+
+	public void baseHeader()
+	{
+		this.WriteByte(PROTOCOL_VERSION);
+	}
+
+	public void Dispatch()
+	{
+		char sDump[MAX_BUFFER_LENGTH];
+
+		int iLen = this.Dump(sDump, MAX_BUFFER_LENGTH);
+
+		this.Close();
+
+		if (!SocketIsConnected(g_hSocket))
+			return;
+
+		// Len required
+		// If len is not included, it will stop at the first \0 terminator
+		SocketSend(g_hSocket, sDump, iLen);
+	}
+}
+
+methodmap Request < Header
+{
+	public Request(int ip_address, StringMap() keyvals)
+	{
+		BaseMessage mreq = BaseMessage();
+		StringMapSnapshot ss = keyvals.Snapshot(); 
+
+		mreq.baseHeader();
+		mreq.WriteInt(ip_address);
+		mreq.WriteInt(keyvals.Size);
+		
+		for(int i = 0; i < ss.Length; i++){
+			ss.getKey(i, keyBuf);

```

Comment:
keyBuf doesn't seem to be declared anywhere, getKey also follows CamelCase, most SP function signature follows CamelCase.